core/Pos/Core/Address.hs

```diff
@@ -346,6 +347,46 @@ deriveLvl2KeyPair ibea scp passphrase wsKey accountIndex addressIndex = do
     -- We don't need to check passphrase twice
     createHDAddressH ibea (ShouldCheckPassphrase False) passphrase hdPass wKey [accountIndex] addressIndex
 
+-- | Derives a HD wallet address and private key from a root key and a path
+-- which is 3 levels deep (i, j, k).
+deriveLvl3KeyPair
+    :: Bi Address'
+    => IsBootstrapEraAddr
+    -> ShouldCheckPassphrase
+    -> PassPhrase
+    -> EncryptedSecretKey
+    -> (Word32, Word32, Word32)
+    -> Maybe (Address, EncryptedSecretKey)
+deriveLvl3KeyPair isBootstrap checkPass passphrase rootKey (i, j, k) =
```

Comment:
What about renaming those variables into something more domain-specific, like `(accountIndex, chainIndex, addressIndex)`?

Reply:
Updated in latest commit.